dask/blockwise.py

```diff
@@ -423,8 +423,8 @@ def __init__(
             name = dep
             if isinstance(dep, BlockwiseDep):
                 name = tokenize(dep)
-                self.io_deps[name] = dep
-                self.numblocks[name] = dep.numblocks
+                self.io_deps[name] = dep  # type: ignore
```

Comment:
This can be fixed by removing `name=dep` above, and having an `else` block below where `name = dep`. Then at all times `name` will be a string.